tests/integration/llm/sagemaker-endpoint-tests.py

```diff
@@ -103,6 +130,67 @@
 DEFAULT_PAYLOAD = {"inputs": "Deep Learning is"}
 
 
+def _upload_metrics(data):
+    cw = boto3.client('cloudwatch')
+    cw.put_metric_data(Namespace='LLM',
+                       MetricData=[{
+                           'MetricName': f"{data['metric_name']}-throughput",
+                           'Unit': 'Count/Second',
+                           'Value': data['throughput']
+                       }, {
+                           'MetricName': f"{data['metric_name']}-avg",
+                           'Unit': 'Milliseconds',
+                           'Value': data['avg']
+                       }, {
+                           'MetricName': f"{data['metric_name']}-p50",
+                           'Unit': 'Milliseconds',
+                           'Value': data['p50']
+                       }, {
+                           'MetricName': f"{data['metric_name']}-p90",
+                           'Unit': 'Milliseconds',
+                           'Value': data['p90']
+                       }, {
+                           'MetricName': f"{data['metric_name']}_p99",
+                           'Unit': 'Milliseconds',
+                           'Value': data['p99']
+                       }])
+
+
+def _get_metric_name(name, config):
+    engine_name = config.get("metrics_config").get("engine")
+    num_partitions = 1
+    model_kwargs = config.get('model_kwargs')
+    if model_kwargs:
+        num_partitions = model_kwargs.get(
+            'number_of_partitions') or model_kwargs.get(
+                "tensor_parallel_degree")
+
+    return f"{name}-{engine_name}-{num_partitions}p"
+
+
+def _run_benchmarks(predictor, config, model_name):
+    latencies = []
+    iterations = 100
+    begin = time.time()
+
+    for _ in range(iterations):
+        start = time.time()
+        predictor.predict(config.get("payload", DEFAULT_PAYLOAD))
+        latencies.append((time.time() - start) * 1000)
+
+    elapsed = (time.time() - begin) * 1000
+
+    benchmark_data = {}
+    benchmark_data['metric_name'] = _get_metric_name(model_name, config)
+    benchmark_data['throughput'] = (iterations - 10) / elapsed * 1000
```

Comment:
elapsed here is the total time for 100 requests, but we're reducing iterations by 10 which seems weird.

Should we have a warmup loop of 10 requests where we don't capture any data, and then an actual benchmark loop of 100 iterations?

Reply:
Also, in the future it would be great if we can parallelize this a bit to exercise the multi worker setup. Doesn't need to be this PR.